src/Intelligence.cpp

```diff
@@ -20,10 +20,10 @@ bool Intelligence::setStartPosition(int x, int y)
     bool isSuccessStartPosition = false;
 
     if(
-            nBoard::BOARD_SIZE >= x
-            &&
-            nBoard::BOARD_SIZE >= y
-            )
+       nBoard::BOARD_SIZE >= x
+       &&
+       nBoard::BOARD_SIZE >= y
+       )
     {
         startPosition.first = x;
         startPosition.second = y;
```

Comment:
If i understand you right, 'x' and 'y' it's a coordinates at board and it's index of array. You must remembers, if size of array is 5, max index should be 4. In your checking results  if x == 5, you set: x = 5. But array[5] will called out of range.
Thanks.